### PR TITLE
feat(snackbar): create custom snackbar

### PR DESCRIPTION
**Pull request by Floberrot**
custom color, variant and position for example

### CODE REVIEW

File: src/components/QSnackBar/QSnackBar.vue [hunk 1]
```diff
@@ -0,0 +1,235 @@
+<!-- eslint-disable vue/multi-word-component-names -->
+<template>
+  <div :class="{ 'q-animate-fadeIn': isOpen, 'q-animate-fadeOut': !isOpen}">
+    <div v-if="isOpen" :class="allClasses">
+      <q-row>
+        <q-col :md="10">
+          <p class="text-snackbar" :class="colorTextClass">{{ message }}</p>
```
Comment (ChxGuillaume) [inline src/components/QSnackBar/QSnackBar.vue:7]: why not using a slot?

File: src/components/QSnackBar/QSnackBar.vue [hunk 2]
```diff
@@ -0,0 +1,235 @@
...
+        </q-col>
+        <q-col :md="2">
+          <q-button
+              class="button-snackbar" variant="plain-rounded-full" :color="colorButton" size="small"
+              @click="Close">X
```
Comment (ChxGuillaume) [inline src/components/QSnackBar/QSnackBar.vue:12]: Is the q-btn necessary?

File: src/components/QSnackBar/QSnackBar.vue [hunk 3]
```diff
@@ -0,0 +1,235 @@
...
+      type: String,
+      validator(value) {
+        const validPosition = ['top-center', 'bottom-center', 'top-right', 'top-left', 'bottom-left', 'bottom-right'];
+        return validPosition.includes(value);
+      },
+    },
+    open: {
+      type: Boolean,
+      default: false,
+    },
```
Comment (ChxGuillaume) [inline src/components/QSnackBar/QSnackBar.vue:43]: Should be required.

Should be V-Model (https://vuejs.org/api/built-in-directives.html#v-model)

File: src/components/QSnackBar/QSnackBar.vue [hunk 4]
```diff
@@ -0,0 +1,235 @@
...
+      default: '#ffffff',
+    },
+    colorText: {
+      type: String,
+      default: '#ffffff',
+      validator(value) {
+        const validColorText = ['white', 'black', 'darkgray'];
+        return validColorText.includes(value);
+      },
+    },
```
Comment (ChxGuillaume) [inline src/components/QSnackBar/QSnackBar.vue:63]: as you should use slots, you shouldn't require textColor

File: src/components/QSnackBar/QSnackBar.vue [hunk 5]
```diff
@@ -0,0 +1,235 @@
...
+    message: {
+      type: String,
+      default: 'Votre message ici.'
+    },
```
Comment (ChxGuillaume) [inline src/components/QSnackBar/QSnackBar.vue:67]: Useles

File: src/components/QSnackBar/QSnackBar.vue [hunk 6]
```diff
@@ -0,0 +1,235 @@
...
+  },
+  data() {
+    return {
+      isOpen: this.open
+    }
```
Comment (ChxGuillaume) [inline src/components/QSnackBar/QSnackBar.vue:72]: ??

https://www.youtube.com/watch?v=-1qju6V1jLM

File: src/components/QSnackBar/QSnackBar.vue [hunk 7]
```diff
@@ -0,0 +1,235 @@
...
+      }
+      if (this.position === 'bottom-center') {
+        allClasses.push('snackbar--bottom-center');
+      }
+      if (this.position === 'bottom-right') {
+        allClasses.push('snackbar--bottom-right');
+      }
+      if (this.position === 'bottom-left') {
+        allClasses.push('snackbar--bottom-left');
+      }
```
Comment (ChxGuillaume) [inline src/components/QSnackBar/QSnackBar.vue:125]: why specific to top-center and bottom-center ?

Should be
top
bottom
left
right
and the variantes of them (botton-left, bottom-right, top-left, top-right)

File: src/components/QSnackBar/QSnackBar.vue [hunk 8]
```diff
@@ -0,0 +1,235 @@
...
+
+      return allClasses;
+    },
+    colorTextClass() {
+      if (this.colorText === 'black')
+        return 'text text--black'
+      if (this.colorText === 'darkgray')
+        return 'text text--darkgray'
+      return 'text text--white';
+    }
```
Comment (ChxGuillaume) [inline src/components/QSnackBar/QSnackBar.vue:135]: as told above, useless

https://www.youtube.com/watch?v=-1qju6V1jLM

File: src/components/QSnackBar/QSnackBar.vue [hunk 9]
```diff
@@ -0,0 +1,235 @@
...
+  },
+  methods: {
+    Close() {
+      this.isOpen = false;
+    }
+  }
```
Comment (ChxGuillaume) [inline src/components/QSnackBar/QSnackBar.vue:141]: should use vmodel ($emit)

**Review by ChxGuillaume — CHANGES_REQUESTED** [review]
Maybe test you stuff more in your example (button to reopen it etc...)
Please use vModel
Please use Slots

In shorts, please use VUEJS FEATURES!!

File: src/components/QSnackBar/QSnackBar.vue [hunk 10]
```diff
@@ -0,0 +1,235 @@
...
+        <q-col :md="10">
+          <p class="text-snackbar" :class="colorTextClass">{{ message }}</p>
+        </q-col>
+        <q-col :md="2">
+          <q-button
+              class="button-snackbar" variant="plain-rounded-full" :color="colorButton" size="small"
+              @click="Close">X
+          </q-button>
+        </q-col>
+      </q-row>
```
Comment (ChxGuillaume) [inline src/components/QSnackBar/QSnackBar.vue:15]: Why using qGrid ?

**Review by ChxGuillaume — CHANGES_REQUESTED** [review]

**Comment by ChxGuillaume** [comment]
https://vuetifyjs.com/en/components/snackbars/#usage

https://www.youtube.com/watch?v=-1qju6V1jLM

**Review by MaxenceMottard — COMMENTED** [review]
Sorry but I prefer @Laroche-Nicolas's implementation of snackbar : #52.
There was a lack of communication and you both made the same component 😕

**Comment by Floberrot** [comment]
> Maybe test you stuff more in your example (button to reopen it etc...) Please use vModel Please use Slots
> 
> In shorts, please use VUEJS FEATURES!!

Sorry but i'm trying to learn vuejs at the same time, so I can forget some things. You can close it, nicolas create a better snackbar

**Comment by MaxenceMottard** [comment]
There is no problem, it's just a pity to have spent time to make it not merge, I'm sorry 😕

**Pull request closed without merging**